com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -88,14 +98,24 @@ internal void __endSendServerRpc(NetworkSerializer serializer, uint rpcMethodId,
                 return;
             }
 
-            var rpcQueueContainer = NetworkManager.RpcQueueContainer;
+            if (serverRpcParams.Send.UpdateStage == NetworkUpdateStage.Unset)
+            {
+                serverRpcParams.Send.UpdateStage = NetworkUpdateLoop.UpdateStage;
+
+                if (serverRpcParams.Send.UpdateStage == NetworkUpdateStage.Initialization)
+                {
+                    serverRpcParams.Send.UpdateStage = NetworkUpdateStage.EarlyUpdate;
+                }
+            }
```

Comment:
A repeating bit of logic that needs to be encapsulated into a function.

Reply:
This one I agree on... you've already uncovered my biggest weakness. :( I definitely do tend to under-functionalize and copy/paste too much.

Reply:
Done.